Java/core/src/main/java/com/amazon/randomcutforest/sampler/CompactSampler.java

```diff
@@ -403,4 +417,132 @@ private void swapWeights(int a, int b) {
+
+        public T size(int size) {
+            this.size = size;
+            return (T) this;
+        }
+
+        public T weight(float[] weight) {
+            this.weight = weight;
+            return (T) this;
+        }
+
+        public T pointIndex(int[] pointIndex) {
+            this.pointIndex = pointIndex;
+            return (T) this;
+        }
+
+        public T sequenceIndex(long[] sequenceIndex) {
+            this.sequenceIndex = sequenceIndex;
+            return (T) this;
+        }
+
+        public T storeSequenceIndexesEnabled(boolean storeSequenceIndexesEnabled) {
+            this.storeSequenceIndexesEnabled = storeSequenceIndexesEnabled;
+            return (T) this;
+        }
+
+        public T validateHeap(boolean validateHeap) {
+            this.validateHeap = validateHeap;
+            return (T) this;
+        }
+
+        public T seed(long seed) {
+            this.random = new Random(seed);
+            return (T) this;
+        }
+
+        public T random(Random random) {
+            this.random = random;
+            return (T) this;
+        }
+
+        public T maxSequenceIndex(long maxSequenceIndex) {
+            this.maxSequenceIndex = maxSequenceIndex;
+            return (T) this;
+        }
+
+        public T mostRecentLambdaUpdate(long sequenceIndexOfMostRecentLambdaUpdate) {
+            this.sequenceIndexOfMostRecentLambdaUpdate = sequenceIndexOfMostRecentLambdaUpdate;
+            return (T) this;
+        }
+
+        public T initialAcceptFraction(double initialAcceptFraction) {
+            this.initialAcceptFraction = initialAcceptFraction;
+            return (T) this;
+        }
+
+        public T lambda(double lambda) {
+            this.lambda = lambda;
+            return (T) this;
+        }
+
+        public CompactSampler build() {
+            return new CompactSampler(this);
+        }
+    }
+
+    public CompactSampler(Builder<?> builder) {
+        // int sampleSize, double lambda, Random random, boolean
+        // storeSequenceIndexesEnabled,
+        // double initialAcceptFraction) {
+        super();
+        checkArgument(builder.initialAcceptFraction > 0, " the admittance fraction cannot be <= 0");
+        checkArgument(builder.capacity > 0, " sampler capacity cannot be <=0 ");
+
+        this.capacity = builder.capacity;
+        this.storeSequenceIndexesEnabled = builder.storeSequenceIndexesEnabled;
+        this.initialAcceptFraction = builder.initialAcceptFraction;
+        this.random = builder.random;
+        this.lambda = builder.lambda;
+        this.maxSequenceIndex = builder.maxSequenceIndex;
+        this.sequenceIndexOfMostRecentLambdaUpdate = builder.sequenceIndexOfMostRecentLambdaUpdate;
+
+        if (builder.weight != null || builder.pointIndex != null || builder.weight != null
+                || builder.sequenceIndex != null || builder.validateHeap) {
+            checkArgument(builder.weight.length == builder.capacity, " incorrect state");
```

Comment:
These error messages won't be very helpful to users.

Reply:
fixed.